Client/game_sa/CVehicleSA.cpp

```diff
@@ -1351,7 +1351,7 @@ void CVehicleSA::RecalculateHandling()
         }
     }
     // Hydraulics Flag fixing
-    if (pInt->m_upgrades[1] && pInt->m_upgrades[1] == 1087)
+    if (pInt->m_upgrades[0] && pInt->m_upgrades[0] == 1087)
```

Comment:
Doesn't work.
The upgrade can be in any position of `m_upgrades` array.
You need to iterate whole array to check it

Reply:
Oh okay but why the code using array instead of std::map so we should check for the key and value

Reply:
`m_upgrades` contains only 8 elements.  An array is faster than a map when the data count is small

Reply:
so that mean we need to check for all upgrades not only nitro and Hydraulics?

Reply:
I'm not @TheNormalnij, but I suppose that means you are supposed to do a linear search over the `m_upgrades` array (i.e., iterate over its elements to find if any of them matches a nitro or hydraulics upgrade) to check if hydraulics and nitro upgrades are present.